File: code/Python-spider/reverse/PBKDF2/pbkdf2.py
# -*- coding: utf-8 -*-
# @Time    : 2023/5/8 14:08
# @Author  : PFinal南丞
# @Email   : lampxiezi@163.com
# @File    : pbkdf2.py
# @Software: PyCharm
import binascii

from Cryptodome.Hash import SHA1
from Cryptodome.Protocol.KDF import PBKDF2

text = 'I lov Python'
salt = b'12345d'

result = PBKDF2(text, salt, count=10, hmac_hash_module=SHA1)
result = binascii.hexlify(result)
print(result)

File: code/Python-spider/reverse/PBKDF2/pbkdf2.js
let CryptoJS = require('crypto-js')

function pbkdf2Encrypt() {
    let text = 'I lov Python'
    let salt = '12345d'
    let encryptedData = CryptoJS.PBKDF2(text, salt, {keySize: 128 / 32, iterations: 10})
    return encryptedData.toString()
}

console.log(pbkdf2Encrypt()) // 22192e6ab76569b73bf0c3e20a9e03df

